Building the assembler code in libavcodec/x86/cabac.h is broken for i386
with GCC 12.1.
Remove this patch once this is fixed either in ffmpeg or in GCC.

diff -urN ffmpeg-4.4.2/libavcodec/x86/cabac.h.orig ffmpeg-4.4.2/libavcodec/x86/cabac.h
--- ffmpeg-4.4.2/libavcodec/x86/cabac.h.orig	2022-05-22 15:09:32.786149500 +0200
+++ ffmpeg-4.4.2/libavcodec/x86/cabac.h	2022-05-22 18:35:06.885673200 +0200
@@ -29,7 +29,7 @@
 #include "libavutil/x86/asm.h"
 #include "config.h"
 
-#if   (defined(__i386) && defined(__clang__) && (__clang_major__<2 || (__clang_major__==2 && __clang_minor__<10)))\
+#if   (defined(__i386) && (!defined(__clang__) || (defined(__clang__) && (__clang_major__<2 || (__clang_major__==2 && __clang_minor__<10)))))\
    || (                  !defined(__clang__) && defined(__llvm__) && __GNUC__==4 && __GNUC_MINOR__==2 && __GNUC_PATCHLEVEL__<=1)\
    || (defined(__INTEL_COMPILER) && defined(_MSC_VER))
 #       define BROKEN_COMPILER 1
